src/app/portlets/content-types/content-types-resolver.service.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { ContentTypeResolver } from './content-types-resolver.service';
+import { TestBed, async } from '@angular/core/testing';
+import { ContentTypesInfoService } from '../../api/services/content-types-info';
+import { CrudService } from '../../api/services/crud';
+import { LoginService } from 'dotcms-js/dotcms-js';
+import { ActivatedRouteSnapshot, ParamMap } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import { LoginServiceMock } from '../../test/login-service.mock';
+import { RouterTestingModule } from '@angular/router/testing';
+import { inject } from '@angular/core/testing';
+import { DotRouterService } from '../../api/services/dot-router-service';
+
+class CrudServiceMock {
+    getDataById() {}
+}
+
+class DotRouterServiceMock {
+    gotoPortlet() {}
+}
+
+const activatedRouteSnapshotMock: any = jasmine.createSpyObj<
+    ActivatedRouteSnapshot
+>('ActivatedRouteSnapshot', ['toString']);
+activatedRouteSnapshotMock.paramMap = {};
+
+describe('ContentTypeResolver', () => {
+    let crudService: CrudService;
+    let router: ActivatedRouteSnapshot;
+    let contentTypeResolver: ContentTypeResolver;
+    let dotRouterService: DotRouterService;
+
+    beforeEach(
+        async(() => {
+            TestBed.configureTestingModule({
+                providers: [
+                    ContentTypeResolver,
+                    ContentTypesInfoService,
+                    {
+                        provide: DotRouterService,
+                        useClass: DotRouterServiceMock
+                    },
+                    { provide: CrudService, useClass: CrudServiceMock },
+                    { provide: LoginService, useClass: LoginServiceMock },
+                    {
+                        provide: ActivatedRouteSnapshot,
+                        useValue: activatedRouteSnapshotMock
+                    }
+                ],
+                imports: [RouterTestingModule]
+            });
+            crudService = TestBed.get(CrudService);
+            router = TestBed.get(ActivatedRouteSnapshot);
+            contentTypeResolver = TestBed.get(ContentTypeResolver);
+            dotRouterService = TestBed.get(DotRouterService);
+        })
+    );
+
+    it('should get and return a content type', () => {
+        activatedRouteSnapshotMock.paramMap.get = () => '123';
+        spyOn(crudService, 'getDataById').and.returnValue(Observable.of({
+            fake: 'content-type',
+            object: 'right?'
+        }));
+        contentTypeResolver
+            .resolve(activatedRouteSnapshotMock)
+            .subscribe((fakeContentType: any) => {
+                expect(fakeContentType).toEqual({
+                    fake: 'content-type',
+                    object: 'right?'
```

Comment:
are you sure that this expect always is executed?, I think you need a tick or a whenStable o something like that

Reply:
https://github.com/dotCMS/core-web/pull/447#discussion_r144920884